src/ray/gcs/gcs_server/gcs_actor_distribution.h

```diff
@@ -0,0 +1,468 @@
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#pragma once
+
+#include <memory>
+
+#include "absl/container/flat_hash_map.h"
+#include "absl/container/flat_hash_set.h"
+#include "ray/common/bundle_spec.h"
+#include "ray/common/id.h"
+#include "ray/common/status.h"
+#include "ray/common/task/scheduling_resources.h"
+#include "ray/common/task/task_spec.h"
+#include "ray/gcs/gcs_server/gcs_actor_manager.h"
+#include "ray/gcs/gcs_server/gcs_actor_scheduler.h"
+#include "ray/gcs/gcs_server/gcs_node_manager.h"
+#include "ray/gcs/gcs_server/gcs_resource_manager.h"
+#include "ray/gcs/gcs_server/gcs_resource_scheduler.h"
+#include "ray/gcs/gcs_server/gcs_table_storage.h"
+#include "ray/util/event_label.h"
+#include "ray/util/resource_util.h"
+#include "src/ray/protobuf/gcs.pb.h"
+
+namespace ray {
+using rpc::Language;
+namespace gcs {
+
+struct GcsJobConfig {
+  explicit GcsJobConfig(const JobID &job_id, uint32_t num_java_workers_per_process,
+                        uint32_t num_initial_java_worker_processes,
+                        uint64_t java_worker_process_default_memory_units,
+                        uint64_t total_memory_units)
+      : job_id_(job_id) {
+    if (num_java_workers_per_process != 0) {
+      num_java_workers_per_process_ = num_java_workers_per_process;
+    }
+
+    num_initial_java_worker_processes_ = num_initial_java_worker_processes;
+
+    if (java_worker_process_default_memory_units > 0) {
+      java_worker_process_default_memory_units_ =
+          java_worker_process_default_memory_units;
+    }
+
+    if (total_memory_units > 0) {
+      total_memory_units_ = total_memory_units;
+    }
+  }
+
+  GcsJobConfig(const JobID &job_id) : job_id_(job_id) {}
+
+  std::string ToString() const {
+    std::ostringstream ostr;
+    ostr << "{ job_id: " << job_id_
+         << ", num_java_workers_per_process: " << num_java_workers_per_process_
+         << ", num_initial_java_worker_processes: " << num_initial_java_worker_processes_
+         << ", java_worker_process_default_memory_gb: "
+         << FromMemoryUnitsToGiB(java_worker_process_default_memory_units_)
+         << ", python_worker_process_default_memory_gb: "
+         << FromMemoryUnitsToGiB(python_worker_process_default_memory_units_)
+         << ", total_memory_gb: " << FromMemoryUnitsToGiB(total_memory_units_) << " }";
+    return ostr.str();
+  }
+
+  JobID job_id_;
+  // The number of workers per worker process.
+  uint32_t num_java_workers_per_process_ = 1;
+  // The number of initial java worker processes of the job.
+  uint32_t num_initial_java_worker_processes_ = 0;
+  // The default memory of java worker process in memory units.
+  uint64_t java_worker_process_default_memory_units_ = 5;
```

Comment:
does it mean capacity/limit? the name is a bit confusing

Reply:
It is indeed the default memory capacity of a java worker process. The name contains `_memory_units_` because it is more straightforward to make us exactly how the memory size.